internal/paascharm/gogen.go

```diff
@@ -0,0 +1,78 @@
+package paascharm
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"text/template"
+)
+
+// CreateGoStructs reads the charmcraft.yaml file from the charmcraftDir directory
+// and outputs the Go file with the proper structs in the outputFile location,
+// using as package name packageName. It will override the outputFile file if
+// it exists and create all the parent directories if missing.
+func CreateGoStructs(charmcraftDir string, packageName string, outputFile string) (err error) {
+	yamlFile, err := os.ReadFile(path.Join(charmcraftDir, CharmcraftFileName))
+	if err != nil {
+		return fmt.Errorf("cannot read charmcraft.yaml file: %v", err)
+	}
+
+	charmcraftConfig, err := ParseCharmcraftYaml(yamlFile)
+	if err != nil {
+		return
+	}
+
+	goStructInfo, err := NewGoStructsData(packageName, charmcraftConfig)
+	if err != nil {
+		return
+	}
+
+	goStructs, err := GenerateGoStructs(goStructInfo)
+	if err != nil {
+		return fmt.Errorf("cannot generate go structs info: %v", err)
+	}
+
+	err = os.MkdirAll(filepath.Dir(outputFile), 0755)
+	if err != nil {
+		return fmt.Errorf("cannot create parent directories for output file %s: %v", outputFile, err)
+	}
+
+	err = os.WriteFile(outputFile, goStructs, 0644)
```

Comment:
I'm wondering if we shouldn't print a warning when we override the file, just to inform the user that the file previously existed and we have overridden it

Reply:
Added a log if the file exists so the user can see the file was overridden.